### PR TITLE
WIP: Using DGL instead of Torch Geometric

### PR DESCRIPTION
**Pull request by gennarinoos**
- [ ] Update and fix unit tests
- [x] Fix CI environment setup
- [ ] Check if point sampler probabilities are negative
- [ ] Double check arch changes (graphconv instead of devconv)
- [x] Re-enable surface distance loss
- [x] Re-enable surface distance loss
- [ ] Double check algorithm complies with the one on the paper

### CODE REVIEW

File: src/neural_mesh_simplification/losses/edge_crossing_loss.py [hunk 1]
```diff
@@ -48,32 +59,15 @@ def find_nearest_triangles(
...
-            else:
-                nearest.append(neighbors[: self.k - 1])
-
-        # Return tensor with consistent shape
-        if len(nearest) > 0 and all(len(n) == 0 for n in nearest):
-            nearest = torch.empty((len(nearest), 0), dtype=torch.long)
-        else:
-            nearest = torch.stack(nearest)
-        return nearest
+        g_knn = dgl.knn_graph(centroids, k)
```
Comment (martinnormark) [inline src/neural_mesh_simplification/losses/edge_crossing_loss.py:62]:
```suggestion
        g_knn = dgl.knn_graph(centroids, k, exclude_self=True)
```

If we want to avoid the triangle itself ranking as the closest, `exclude_self` should be set. We explicitly removed self-connections prior to this change.

File: tests/losses/test_edge_crossings_loss.py [hunk 2]
```diff
@@ -38,7 +39,7 @@ def test_find_nearest_triangles(loss_fn):
 
     nearest = loss_fn.find_nearest_triangles(vertices, faces)
     assert nearest.shape[0] == faces.shape[0]
-    assert nearest.shape[1] == 1  # k-1 = 1, since k=2
+    assert nearest.shape[1] == k_val
```
Comment (martinnormark) [inline tests/losses/test_edge_crossings_loss.py:42]: I believe the comment from before does that fit anymore. The function will only return less than `k` if the number of centroids are less than `k`.
